### PR TITLE
Remove getter override method

### LINKED ISSUE #6: Relationship cache causes unexpected behaviour
Hi @excid3, thanks for forking it and taking a stab on maintaining acts_as_tenant.

I introduced tenancy to an app early this year and have encountered unexpected behaviour due to this override: 

https://github.com/excid3/acts_as_tenant/blob/master/lib/acts_as_tenant/model_extensions.rb#L177-L183

The gains are negligible, less than 0.5ms, and ActiveRecord caches that if we're calling the tenant object over and over with the same object.

If the tenant object changed/updated underneath between when it was first loaded to when the association is called, it causes to return a stale object. That's pretty much what I've hit a few times until forking and dropping these lines: https://github.com/fivegoodfriends/acts_as_tenant/commit/33ac6cfee2e339f3883cdd945e0df00297ff959d

By letting `belongs_to` work as intended, it hits the db, taking less than 0.5ms, and AR caches it for free after that.

At the time I wrote the initial [issue](https://github.com/ErwinM/acts_as_tenant/issues/226) at ErwinM's repo, I didn't find the reason behind the caching through looking commit history.

Is there a real need for it you may be aware of?
Would you consider accepting a PR that drops it?
I'd be happy to do the work.

I appreciate your help and time.

### PR DESCRIPTION
**Pull request by bernardeli**
There are negligible performance gains in caching the tenant object to
justify the complexity.

Removing the cache lets ActiveRecord return the fresh object and it uses
internal caching when accessing it from the same object.

@excid3 Let me know what you think. Thanks a bunch.

Closes #6.

### CODE REVIEW

**Comment by bernardeli** [comment]
I've broken something that I did not expect 🤔. Looking at that now.

**Comment by bernardeli** [comment]
Sharing some findings here.

The change I made revealed something funky going on. The failure is about the foreign key being Account#name.

After some debugging, I saw in `spec/debug.log` the following line:

```
D, [2020-11-13T13:47:27.464317 #63189] DEBUG -- :   Account Create (0.3ms)  INSERT INTO "accounts" DEFAULT VALUES
```

The values are not getting into the database at all, hence the failure.

Interestingly, it happens only when I run the whole suite. Running the single file works just fine. I suspected interdependency between test files. That can be seen by asking rspec to run the suite in random order. I'll see if I can tackle that, too.

**Comment by bernardeli** [comment]
Ok cool. It was another test file rewriting the Account class. The order it was loaded affected the entire suite. Changes are in b783280.

**Comment by bernardeli** [comment]
I went a bit overboard with the idea of removing multiple Account classes throughout the tests after I found how it breaks.

This commit 86be11b5df96afe6300d83c6581bbbea3fdccaa3 moves the active record test models one layer above, which makes them required to run all tests. It sounds a sensible change since ActsAsTenant relies on a record to be fetched from the database.

Let me know what you think of these changes. I appreciate very much any feedback you may have, @excid3.

Cheers, Ricardo.

**Comment by excid3** [comment]
You've been busy, thanks @bernardeli 🙏 

Reviewing now.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 $LOAD_PATH.unshift(File.join(File.dirname(__FILE__), "..", "lib"))
 $LOAD_PATH.unshift(File.dirname(__FILE__))
 
-require "active_record_helper"
+require "rails/all"
+require "rspec/rails"
```
Comment (excid3) [inline spec/spec_helper.rb:5]: This line gives me trouble which is why I moved it in master to after the Rails app exists.

```ruby
An error occurred while loading ./spec/acts_as_tenant/test_tenant_middleware_spec.rb.
Failure/Error: require "rspec/rails"

RuntimeError:
  missing run or map statement
# ./spec/spec_helper.rb:5:in `<top (required)>'
# ./spec/acts_as_tenant/test_tenant_middleware_spec.rb:1:in `<top (required)>'
```

**Comment by excid3** [comment]
Looks good other than the rspec-rails issue I had (which is fixed in master). Can you change that or update this PR so I can edit files?

Next thing I want to do is refactor that spec_helper stuff into a standard dummy Rails app for testing.

**Comment by excid3** [comment]
Actually just gonna merge and fix in master. 👍

**Comment by bernardeli** [comment]
Thanks a bunch, @excid3. I will point the Gemfile to this fork from now on.

This gem is an important part of the @fivegoodfriends architecture and it's important for me that this gem is in good shape and is well maintained. I'm open to help out whenever I find a suitable time to do so during my work hours. 

Again, thanks for promptly helping me with this PR.

**Comment by excid3** [comment]
Thanks for the help @bernardeli! Will definitely take you up on that. 🙏

**Pull request merged**